ul/tests/association.rs

```diff
@@ -2,9 +2,250 @@ use dicom_dictionary_std::uids::VERIFICATION;
 use dicom_ul::ClientAssociationOptions;
 use rstest::rstest;
 use std::time::Instant;
+use dicom_ul::association::Association;
+#[cfg(feature = "sync-tls")]
+use std::sync::Arc;
+
+#[cfg(feature = "sync-tls")]
+fn ensure_test_certs() -> Result<(), Box<dyn std::error::Error>> {
+    use rustls_cert_gen::CertificateBuilder;
+    use rcgen::SanType;
+    use std::{convert::TryInto, net::IpAddr, str::FromStr, path::PathBuf};
+
+    let out_dir = PathBuf::from(env!("CARGO_MANIFEST_DIR")).join("certs");
+    let cert_names = vec!["ca.pem", "ca.key.pem", "client.pem", "client.key.pem", "server.pem", "server.key.pem"];
+    if cert_names.iter().all(|path| out_dir.join(path).exists()){
+        println!("All certs exist, exiting");
+        return Ok(());
+    }
+    std::fs::remove_dir_all(&out_dir).expect("Could not remove certs dir");
```

Comment:
This function call fails if the directory does not exist, which was the case when running in CI or when cleaning up old artifacts. Something like this makes the tests more reliable, I think.

```suggestion
    if out_dir.exists() {
        std::fs::remove_dir_all(&out_dir).expect("Could not remove certs dir");
    }
```